### PR TITLE
JSDK 118 update SDK to make use of TLS1.2

### PR DESCRIPTION
**Pull request by acostros**

### CODE REVIEW

File: src/com/mambu/apisdk/util/RequestExecutorImpl.java [hunk 1]
```diff
@@ -205,15 +208,30 @@ public ByteArrayOutputStream executeRequest(String urlString, ParamsMap params,
...
 				// set cookies validation on ignore
 				.setDefaultRequestConfig(RequestConfig.custom().setCookieSpec(CookieSpecs.IGNORE_COOKIES).build())
+				.setSSLSocketFactory(createSslConnectionSocketFactory())
 				.build();
 
 		return httpClient;
 	}
 
+	/**
+	 * Creates custom SSLConnectionSocketFactory using and set it to use only the TLSv1.2 as supported protocol
```
Comment (florinparascan) [inline src/com/mambu/apisdk/util/RequestExecutorImpl.java]: using can be removed, I guess

Comment (acostros) [reply src/com/mambu/apisdk/util/RequestExecutorImpl.java]: yup, thanks!

File: src/com/mambu/apisdk/util/RequestExecutorImpl.java [hunk 2]
```diff
@@ -205,15 +208,30 @@ public ByteArrayOutputStream executeRequest(String urlString, ParamsMap params,
...
+	 * Creates custom SSLConnectionSocketFactory and set it to use only the TLSv1.2 as supported protocol
+	 * 
+	 * @return newly created SSLConnectionSocketFactory
+	 */
+	private SSLConnectionSocketFactory createSslConnectionSocketFactory() {
+
+		SSLConnectionSocketFactory sslConnFactory = new
+				SSLConnectionSocketFactory(SSLContexts.createDefault(),
+				new String[] {"TLSv1.2"}, null,
+				SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
```
[the remark this answers is not in the document]
Comment (acostros) [reply src/com/mambu/apisdk/util/RequestExecutorImpl.java]: Yep, it is and we have to stick with it do to mobile app constraints

File: src/com/mambu/apisdk/util/RequestExecutorImpl.java [hunk 3]
```diff
@@ -205,15 +208,30 @@ public ByteArrayOutputStream executeRequest(String urlString, ParamsMap params,
...
+	/**
+	 * Creates custom SSLConnectionSocketFactory and set it to use only the TLSv1.2 as supported protocol
+	 * 
+	 * @return newly created SSLConnectionSocketFactory
+	 */
+	private SSLConnectionSocketFactory createSslConnectionSocketFactory() {
+
+		SSLConnectionSocketFactory sslConnFactory = new
+				SSLConnectionSocketFactory(SSLContexts.createDefault(),
+				new String[] {"TLSv1.2"}, null,
```
Comment (florinparascan) [inline src/com/mambu/apisdk/util/RequestExecutorImpl.java]: "TLSv1.2" can be extracted as a class-level constant. Also, please rename the rest of the constants, so all of them will use uppercase letters. Thank you!

Comment (acostros) [reply src/com/mambu/apisdk/util/RequestExecutorImpl.java]: Done, thanks!

**Pull request merged**